### PR TITLE
[SPARK-6379][SQL] Support a functon to call user-defined functions registered in SQLContext

### PR DESCRIPTION
**Pull request by maropu**
This is useful for using pre-defined UDFs in SQLContext;

val df = Seq(("id1", 1), ("id2", 4), ("id3", 5)).toDF("id", "value")
val sqlctx = df.sqlContext
sqlctx.udf.register("simpleUdf", (v: Int) => v \* v)
df.select($"id", sqlctx.callUdf("simpleUdf", $"value"))

### CODE REVIEW

**Comment by rxin** [comment]
Can't we make this explicit by using "callUDF"?

**Comment by maropu** [comment]
You mean df.select($"id", callUDF($"value"))?

**Comment by rxin** [comment]
I meant something like
```scala
df.select($"id", callUDF("simpleUDF", $"value"))
```

**Comment by maropu** [comment]
Ok, I'll try to refine my codes.

**Comment by maropu** [comment]
@rxin fixed, and plz check it.

File: sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala [hunk 1]
```diff
@@ -212,6 +212,22 @@ class SQLContext(@transient val sparkContext: SparkContext)
...
+   * {{{
+   *  import org.apache.spark.sql._
+   *
+   *  val df = Seq(("id1", 1), ("id2", 4), ("id3", 5)).toDF("id", "value")
+   *  val sqlctx = df.sqlContext
+   *  sqlctx.udf.register("simpleUdf", (v: Int) => v * v)
+   *  df.select($"id", sqlctx.callUdf("simpleUdf", $"value"))
+   * }}}
+   */
+  def callUdf(udfName: String, cols: Column*) =
```
Comment (marmbrus) [inline sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala]: This probably belongs in `functions.scala` with the other `callUDF` methods.  Also, please explicitly specify return types for public methods.

**Comment by marmbrus** [comment]
Once you have updated, can you also update the PR description, as this becomes the commit message.  Thanks!

**Comment by maropu** [comment]
The description updated and the patch fixed.

File: sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala [hunk 2]
```diff
@@ -212,6 +212,22 @@ class SQLContext(@transient val sparkContext: SparkContext)
...
+   * {{{
+   *  import org.apache.spark.sql._
+   *
+   *  val df = Seq(("id1", 1), ("id2", 4), ("id3", 5)).toDF("id", "value")
+   *  val sqlctx = df.sqlContext
+   *  sqlctx.udf.register("simpleUdf", (v: Int) => v * v)
+   *  df.select($"id", sqlctx.callUdf("simpleUdf", $"value"))
+   * }}}
+   */
+  def callUdf(udfName: String, cols: Column*): Column =
```
Comment (marmbrus) [inline sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala]: > This probably belongs in functions.scala with the other callUDF methods.

This is still defined in the wrong file.

File: sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala [hunk 3]
```diff
@@ -212,6 +212,22 @@ class SQLContext(@transient val sparkContext: SparkContext)
   val udf: UDFRegistration = new UDFRegistration(this)
 
   /**
+   * Call an user-defined function which is registered
+   * in functionRegistry.
```
Comment (marmbrus) [inline sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala]: Nit: I don't think this this needs to be wrapped.

File: sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala [hunk 4]
```diff
@@ -212,6 +212,22 @@ class SQLContext(@transient val sparkContext: SparkContext)
...
+   * Example:
+   * {{{
+   *  import org.apache.spark.sql._
+   *
+   *  val df = Seq(("id1", 1), ("id2", 4), ("id3", 5)).toDF("id", "value")
+   *  val sqlctx = df.sqlContext
+   *  sqlctx.udf.register("simpleUdf", (v: Int) => v * v)
+   *  df.select($"id", sqlctx.callUdf("simpleUdf", $"value"))
```
Comment (marmbrus) [inline sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala]: No `sqlCtx.` once this is moved.

**Comment by maropu** [comment]
I'll fix soon.

File: sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala (hunk 2 above)
Comment (maropu) [inline sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala]: Oh, misunderstood.
I'll move the function into `functions.scala`.

**Comment by marmbrus** [comment]
ok to test

**Comment by marmbrus** [comment]
Thanks!  Merged to master.

**Pull request closed without merging**